src/Fixer/Whitespace/AlignEnumValuesFixer.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Whitespace;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+use PhpCsFixer\Tokenizer\TokensAnalyzer;
+
+final class AlignEnumValuesFixer extends AbstractFixer
```

Comment:
IMHO this rules cannot be added because it conflicts with BinaryOperatorFixer.

What if I use BinaryOperatorFixer with single_space strategy and this rule together ? The file will never be fixed.

The BinaryOperatorFixer need to ignore enums then.
But doing such a change will be a BC-break.

Reply:
This rule needs to run after BinaryOperatorFixer (so we'll need to work on the priority) as it is much more specific. No need to review BinaryOperatorFixer IMO.

Reply:
> This rule needs to run after BinaryOperatorFixer (so we'll need to work on the priority) as it is much more specific. No need to review BinaryOperatorFixer IMO.

This would means that everytime you run php-cs-fixer, it will 
- Change the code because of BinaryOperatorFixer
- Change the ode back because of this rule

The generated code is supposed to validate every single rules after one run. This won't be the case here.

Reply:
Ok, I didn't know that. I thought that rules were to run one after the other, hence the priority.

What are the solutions then?

I start to think the BinaryOperator rule looks like a huge monster that wants to do too much 😄 since it crosses the path of array, enum & match alignment...

Reply:
`conflicts with BinaryOperatorFixer` I wonder, should the `BinaryOperatorFixer` dix `=>` of enum, like is `=>` in that case a binary operator, I think maybe not.

Reply:
FYI: It does not conflict with `binary_operator_spaces` with `at_least_single_space` strategy, added in #7175.